data-processing/entities/terms/extractor.py

```diff
@@ -10,110 +9,169 @@
+			for object in terms_and_def_text_html:
+				term_list = re.split("\s+", object[0])
+				if len(term_list) > 1:
+					if object[0] in text_segment_string:
+						matches_tuples.append([object[0], text_segment_string, object[1], object[2]])
+
+			for i in matches_tuples:
+				list_to_add = i[:-1]
+				if isinstance(i[-1], list):
+					last_elem = tuple(i[-1])
+				else:
+					last_elem = i[-1]
+				list_to_add.append(last_elem)
+				tuple_to_add = tuple(list_to_add)
+				matches.add(tuple_to_add)
+
+			###Getting the start and end indices of the matched terms
+			for match in matches:
+				start_index_of_string = text_segment_string.find(match[0])
+				end_index_of_string = start_index_of_string + len(match[0]) - 1
+				matched_terms.append(match[0])
+				matched_terms_context_str.append(match[1])
+				matched_definition_texts.append(match[2])
+				matched_definition_htmls.append(match[3])
+				matched_terms_start.append(text_segment.tex_start + start_index_of_string)
+				matched_terms_end.append(text_segment.tex_start + end_index_of_string)
+
+		def duplicates(lst, item):
+			return [i for i, x in enumerate(lst) if x == item]
+
+		# To Do:
+		# 1) Determine the objects that share the same start index.
+		# 2) Keep the object with the largest end index.
+		# 3) Remove all other objects from all lists.
+
+		def delete_duplicate_start(term, start, end, context_str):
+			for start_index in start:
+
+				# 1)
+				duplicate_indices = duplicates(start, start_index)
+
+				# 2) Keep the object with the largest end index.
+				if len(duplicate_indices) > 1:
+					print("OH NO")
+					ending_indices = []
+					for duplicate_index in duplicate_indices:
+						ending_indices.append(matched_terms_end[duplicate_index])
+
+					# get the index of the largest ending index
+					index_of_largest_ending_index = ending_indices.index(max(ending_indices))
+					duplicate_index_to_keep = index_of_largest_ending_index
+
+					# remove the index that we are going to keep from the list of duplicate indices.
+					duplicate_indices.pop(duplicate_index_to_keep)
+
+					# remove the objects that are located at the remaining indices from all lists.
+					for index in duplicate_indices:
+						matched_terms.pop(index)
+						matched_terms_start.pop(index)
+						matched_terms_end.pop(index)
+						matched_terms_context_str.pop(index)
+						matched_definition_texts.pop(index)
+						matched_definition_htmls.pop(index)
+
+		delete_duplicate_start(matched_terms, matched_terms_start, matched_terms_end, matched_terms_context_str)
+
+		index = 0
+		for term_object_number in range(len(matched_terms)):
+			yield Term(
+				name=matched_terms[term_object_number],
+				defintion_text=matched_definition_texts[term_object_number],
+				definition_html=matched_definition_htmls[term_object_number],
+				# start position of the term in the TeX
+				start=matched_terms_start[term_object_number],
+				end=matched_terms_end[term_object_number],  # end position
```

Comment:
nit pick: Axe these comments, unless they convey something new that isn't inferable from the code or lending additional documentation beyond what's already there for the SerializableEntity class.

```suggestion
				end=matched_terms_end[term_object_number],
```